src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1123,7 +1125,10 @@ private function addSerializerSection(ArrayNodeDefinition $rootNode, callable $e
                         ->arrayNode('default_context')
                             ->normalizeKeys(false)
                             ->useAttributeAsKey('name')
-                            ->defaultValue([])
+                            ->defaultValue($this->debug && class_exists(JsonParser::class)
+                                ? [JsonDecode::DETAILED_ERROR_MESSAGES => true]
+                                : [],
+                            )
```

Comment:
wouldn't it be better to do this in the extension, so that people that configure a default context can still get the new default?

Reply:
This works too in such case. Thanks to using hashmap instead of list and the way how symfony/config merge things, similarly as this works
```yaml
framework:
  foo:
when@dev:
  framework:
    bar:
```
would in @dev result into
```yaml
framework:
  foo:
  bar:
```

Reply:
Can we add functional tests that makes sure the default is properly merged into the user-provided context?

Reply:
Added. And indeed, can't use defaultValue() for this.